SU2_CFD/include/numerics/CNumerics.hpp

```diff
@@ -0,0 +1,1302 @@
+   * \param[in] val_normal - Normal vector, the norm of the vector is the area of the face.
+   * \param[out] val_invp_tensor - Pointer to inverse of the P matrix.
+   */
+  void GetPMatrix_inv(su2double *val_density, su2double *val_velocity,
+                      su2double *val_soundspeed, su2double *val_normal,
+                      su2double **val_invp_tensor);
+
+  /*!
+   * \brief Compute viscous residual and jacobian.
+   */
+  void GetAdjViscousFlux_Jac(su2double Pressure_i, su2double Pressure_j, su2double Density_i, su2double Density_j,
+                             su2double ViscDens_i, su2double ViscDens_j, su2double *Velocity_i, su2double *Velocity_j,
+                             su2double sq_vel_i, su2double sq_vel_j,
+                             su2double XiDens_i, su2double XiDens_j, su2double **Mean_GradPhi, su2double *Mean_GradPsiE,
+                             su2double dPhiE_dn, su2double *Normal, su2double *Edge_Vector, su2double dist_ij_2, su2double *val_residual_i,
+                             su2double *val_residual_j,
+                             su2double **val_Jacobian_ii, su2double **val_Jacobian_ij, su2double **val_Jacobian_ji,
+                             su2double **val_Jacobian_jj, bool implicit);
+
+  /*!
+   * \brief Computation of the projected inviscid lambda (eingenvalues).
+   * \param[in] val_velocity - Value of the velocity.
+   * \param[in] val_soundspeed - Value of the sound speed.
+   * \param[in] val_normal - Normal vector, the norm of the vector is the area of the face.
+   * \param[in] val_Lambda_Vector - Pointer to Lambda matrix.
+   */
+  void GetJacInviscidLambda_fabs(su2double *val_velocity, su2double val_soundspeed,
+                                 su2double *val_normal, su2double *val_Lambda_Vector);
+
+  /*!
+   * \brief Compute the numerical residual.
+   * \param[out] val_residual - Pointer to the total residual.
+   * \param[in] config - Definition of the particular problem.
+   */
+  inline virtual void ComputeResidual(su2double *val_residual, CConfig* config) { }
+
+  /*!
+   * \overload
+   * \param[out] val_residual - Pointer to the total residual.
+   * \param[out] val_Jacobian_i - Jacobian of the numerical method at node i (implicit computation).
+   * \param[out] val_Jacobian_j - Jacobian of the numerical method at node j (implicit computation).
+   * \param[in] config - Definition of the particular problem.
+   */
+  inline virtual void ComputeResidual(su2double *val_residual, su2double **val_Jacobian_i,
+                                      su2double **val_Jacobian_j, CConfig* config) { }
+
+  /*!
+   * \overload For numerics classes that store the residual/flux and Jacobians internally.
+   * \param[in] config - Definition of the particular problem.
+   * \return A lightweight const-view (read-only) of the residual/flux and Jacobians.
+   */
+  inline virtual ResidualType<> ComputeResidual(const CConfig* config) { return ResidualType<>(nullptr,nullptr,nullptr); }
```

Comment:
This is the overload for that return type. You will also notice that following #853 I am making the use of CConfig read-only whenever possible in low level classes such as CNumerics.